src/pyclts/commands/map.py

```diff
@@ -1,96 +1,121 @@
+                        sound2 = bipa[raw_grapheme[match[1].start() :]]
+                        if sound1.type == "consonant" and sound2.type == "consonant":
+                            # check for prenasalized stuff
+                            if sound1.manner == "nasal" and (
+                                sound2.place == sound2.place
+                                or sound2.manner
+                                in ["stop", "affricate", "fricative", "implosive"]
+                            ):
+                                row["BIPA"] = "*ⁿ" + str(sound2)
+                            else:
+                                row["BIPA"] = "(?)"
                         else:
-                            data[i][bidx] = '?'
-                            count += 1
+                            row["BIPA"] = "<NA>"
                     else:
-                        data[i][bidx] = '<NA>'
-                        count += 1
-                else:
-                    data[i][bidx] = '<NA>'
-                    count += 1
-            elif s.type == 'marker':
-                data[i][bidx] = rg
-            elif s.type == 'cluster':
-                # check for prenasalized stuff
-                if s.from_sound.manner == 'nasal' and (
-                        s.from_sound.place == s.to_sound.place or \
-                                s.to_sound.manner in ['stop', 'affricate',
-                                    'fricative', 'implosive']):
-                    data[i][bidx] = '*ⁿ'+s.to_sound.s
-                elif s.to_sound.manner == 'fricative' and s.from_sound.manner == 'stop':
-                    ns = bipa[s.to_sound.name.replace('fricative',
-                        'affricate')]
-                    if ns.type == 'consonant':
-                        data[i][bidx] = '*'+ns.to_sound.s
+                        row["BIPA"] = "<NA>"
+                elif sound.type == "marker":
+                    row["BIPA"] = raw_grapheme
+                elif sound.type == "cluster":
+                    # check for prenasalized stuff
+                    if sound.from_sound.manner == "nasal" and (
+                        sound.from_sound.place == sound.to_sound.place
+                        or sound.to_sound.manner
+                        in ["stop", "affricate", "fricative", "implosive"]
+                    ):
+                        row["BIPA"] = "*ⁿ" + str(sound.to_sound)
+                    elif (
+                        sound.to_sound.manner == "fricative"
+                        and sound.from_sound.manner == "stop"
+                    ):
+                        new_sound = bipa[
+                            s.to_sound.name.replace("fricative", "affricate")
+                        ]
+                        if new_sound.type == "consonant":
+                            row["BIPA"] = "*" + str(new_sound.to_sound)
+                        else:
+                            row["BIPA"] = "<NA>"
```

Comment:
Are we explicitly MAKING an NA mapping here? As far as I wanted to do this, we would only KEEP NA mappings, but never impose them.

Reply:
I was following the original code. I get it should be mapped to `(?)`, then?